src/gui/dbsettings/DatabaseSettingsWidgetDatabaseKey.cpp

```diff
@@ -177,31 +177,32 @@ bool DatabaseSettingsWidgetDatabaseKey::save()
         return false;
     }
 
-    // Show warning if database password is weak
-    if (!m_passwordEditWidget->isEmpty()
-        && m_passwordEditWidget->getPasswordQuality() < PasswordHealth::Quality::Good) {
-        auto dialogResult = MessageBox::warning(this,
-                                                tr("Weak password"),
-                                                tr("This is a weak password! For better protection of your secrets, "
-                                                   "you should choose a stronger password."),
-                                                MessageBox::ContinueWithWeakPass | MessageBox::Cancel,
-                                                MessageBox::Cancel);
-
-        if (dialogResult == MessageBox::Cancel) {
+    if (!m_passwordEditWidget->isEmpty()) {
+        // Prevent setting password with a quality less than the minimum required
+        auto minQuality = qBound(0, config()->get(Config::Security_DatabasePasswordMinimumQuality).toInt(), 4);
+        if (m_passwordEditWidget->getPasswordQuality() < static_cast<PasswordHealth::Quality>(minQuality)) {
+            MessageBox::critical(this,
+                                 tr("Weak password"),
+                                 tr("The provided password does not meet the minimum quality requirement."),
+                                 MessageBox::Ok,
+                                 MessageBox::Ok);
             return false;
         }
-    }
 
-    // If enforced in the config file, deny users from continuing with a weak password
-    auto minQuality =
-        static_cast<PasswordHealth::Quality>(config()->get(Config::Security_DatabasePasswordMinimumQuality).toInt());
-    if (!m_passwordEditWidget->isEmpty() && m_passwordEditWidget->getPasswordQuality() < minQuality) {
-        MessageBox::critical(this,
-                             tr("Weak password"),
-                             tr("You must enter a stronger password to protect your database."),
-                             MessageBox::Ok,
-                             MessageBox::Ok);
-        return false;
+        // Show warning if database password is weak or poor
+        if (m_passwordEditWidget->getPasswordQuality() < PasswordHealth::Quality::Good) {
```

Comment:
I think you need to take into account whether this is a normal password or a passphrase.

Reply:
Hmmmm, that's an interesting point, but I'm not sure how to proceed

Reply:
Going to table that for now, technically a long passphrase rates as excellent:

![image](https://github.com/keepassxreboot/keepassxc/assets/2809491/aec65653-e606-4f0e-bf97-ac44c3192091)

Three words rates weak, which I would expect:

![image](https://github.com/keepassxreboot/keepassxc/assets/2809491/d364217f-ceec-416e-a47e-4a502ac8d544)